src/components/selector_menu/index.js

```diff
@@ -128,23 +126,30 @@ var SelectorMenu = React.createClass({
       return;
     }
 
-    var visible = _.map(fuzzy.filter(filterBy, this.state.visible), function(result) {
-      return result.string;
-    });
+    var visible = _.pluck(fuzzy.filter(filterBy, this.getOptionNames()), 'string');
 
     this.setState({
       visible: visible
     });
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    if (nextProps.optionsList.length !== this.props.optionsList.length) {
+      this.setState({ visible: [] });
+    }
+  },
+
   render: function() {
     var wrapperClass = this.state.expanded ? 'selector__wrapper expanded' : 'selector__wrapper';
     var innerClass = this.state.expanded ? 'inner-wrapper expanded' : 'inner-wrapper';
+    var visible = this.state.visible.length > 0 ? this.state.visible : this.getOptionNames();
+    var selected = this.state.selected && this.props.defaultSelection === undefined ?
```

Comment:
@wookiehangover Hmm, I think I'm not going to be able to fully understand this issue unless I see the bug in action because I'm kind of missing a) how the form (<search>) itself is relevant (since we're talking about issues around the <label> and <list>) and b) what this means: "when an external change happens (like in the filters UI, where the filters are controlled in 2 different places) it gets stuck in the wrong state" (ie, why setting state in componentWillReceiveProps isn't sufficient to keep these in sync). What branch should I look at to explore?

Also, don't mean to belabor a relatively trivial change, just think there's an opportunity for me to learn here.